src/main/java/core/basesyntax/service/strategy/BalanceOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    private FruitStorageDao storageDao;
+
+    public BalanceOperation(FruitStorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storageDao.getAllFruits().put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
Directly using `put` to update the balance can be risky. Consider checking if the `transaction.getQuantity()` is not negative before updating the balance to prevent incorrect data in the storage.